public/src/api/makeApiRequest.ts

```diff
@@ -7,12 +7,20 @@ export function makeApiRequest<T>(
   const config: AxiosRequestConfig =
     typeof urlOrConfig === 'string' ? { url: urlOrConfig } : urlOrConfig;
 
-  return httpClient.request(config).then((response) => {
-    if (response) {
-      const token = response.headers.authorization;
-      token && sessionStorage.setItem('token', token);
-
-      return response.data;
-    }
-  });
+  return httpClient
+    .request(config)
+    .then((response) => {
+      if (response) {
+        const token = response.headers.authorization;
+        token && sessionStorage.setItem('token', token);
+        return response.data;
+      }
+    })
+    .catch((error) => {
+      if (error.response.status === 401) {
+        sessionStorage.removeItem('email');
```

Comment:
As far as I know, you don't put `email` to session storage at all. There's no reason to remove it.
```suggestion
```